api/src/main/java/com/velocitypowered/api/proxy/player/ResourcePackInfo.java

```diff
@@ -0,0 +1,120 @@
+
+  /**
+   * Gets the SHA-1 hash of the resource-pack
+   * See {@link Builder#setHash(byte[])} for more information.
+   *
+   * @return the hash if present or null otherwise
+   */
+  @Nullable
+  byte[] getHash();
+
+  /**
+   * Gets the {@link Origin} of the resource-pack.
+   *
+   * @return the origin of the resource pack
+   */
+  Origin getOrigin();
+
+  interface Builder {
+
+    /**
+     * Sets the resource-pack as required to play on the network.
+     * This feature was introduced in 1.17.
+     * Setting this to true has one of two effects:
+     * If the client is on 1.17 or newer:
+     *  - The resource-pack prompt will display without a decline button
+     *  - Accept or disconnect are the only available options but players may still press escape.
+     *  - Forces the resource-pack offer prompt to display even if the player has
+     *    previously declined or disabled resource packs
+     *  - The player will be disconnected from the network if they close/skip the prompt.
+     * If the client is on a version older than 1.17:
+     *  - If the player accepts the resource pack or has previously accepted a resource-pack
+     *    then nothing else will happen.
+     *  - If the player declines the resource pack or has previously declined a resource-pack
+     *    the player will be disconnected from the network
+     *
+     * @param shouldForce whether or not to force the client to accept the resource pack
+     */
+    Builder setShouldForce(boolean shouldForce);
+
+    /**
+     * Sets the SHA-1 hash of the provided resource pack.
+     * Note: It is recommended to always set this hash.
+     * If this hash is not set/ not present then the client will always download
+     * the resource pack even if it may still be cached. By having this hash present,
+     * the client will check first whether or not a resource pack by this hash is cached
+     * before downloading.
+     *
+     * @param hash the SHA-1 hash of the resource-pack
+     */
+    Builder setHash(@Nullable byte[] hash);
+
+    /**
+     * Sets a {@link Component} to display on the download prompt.
+     * This will only display if the client version is 1.17 or newer.
+     *
+     * @param prompt the component to display
+     */
+    Builder setPrompt(@Nullable Component prompt);
+
+    /**
+     * Builds the {@link ResourcePackInfo} from the provided info for use with
+     * {@link com.velocitypowered.api.proxy.Player#sendResourcePackOffer(ResourcePackInfo)}.
+     * Note: Some features may be version-dependent. Check before use.
+     *
+     * @return a ResourcePackInfo instance from the provided information
+     */
+    ResourcePackInfo build();
+  }
+
+  /**
+   * Represents the origin of the resource-pack.
+   */
+  enum Origin {
+    /**
+     * Resource-pack originated from the downstream server.
+     */
+    DOWNSTREAM_SERVER,
+    /**
+     * The player declined to download the resource pack.
+     */
+    PLUGIN_ON_PROXY
```

Comment:
Looks like that comment is wrong

Reply:
I know and its fixed, but till https://github.com/VelocityPowered/Velocity/pull/382/files#r640721589 is resolved I wont push a commit to fix it